release/ray_release/reporter/db.py

```diff
@@ -1,18 +1,107 @@
 import time
+import re
 import json
 import boto3
+from typing import List
 from botocore.config import Config
 
 from ray_release.reporter.reporter import Reporter
 from ray_release.result import Result
 from ray_release.config import Test
 from ray_release.logger import logger
 
+CRASH_PATTERN_MAX_LENGTH = 4000
+TRACEBACK_PATTERN = "Traceback (most recent call last)"
+
 
 class DBReporter(Reporter):
     def __init__(self):
         self.firehose = boto3.client("firehose", config=Config(region_name="us-west-2"))
 
+    def compute_crash_pattern(self, logs: str) -> str:
+        stack_trace = self._compute_stack_trace(logs.splitlines())
+        return self._compute_signature(stack_trace)[:CRASH_PATTERN_MAX_LENGTH]
+
+    def _compute_signature(self, stack_trace: List[str]) -> str:
+        """
+        Compute signature pattern from stack trace, by remove factors such as date,
+        time, temp directory, line numbers, etc. This help to aggregate similar logs
+        into same bug patterns
+        """
+        massaged_trace = []
+        for line in stack_trace:
+            line = re.sub(r"\d", "", line.strip())
+            if line == "Traceback (most recent call last):":
+                continue
+            file_line = re.search(r'File "(.*)", (.*)', line)
+            if file_line:
+                line = f'{file_line.group(1).split("/")[-1]}{file_line.group(2)}'
```

Comment:
could you add a comment on what this line is dong?

like: "take the file's base name, and line number"? (not sure if I interpreted correctly..)